### ISSUE TITLE
no-unused-vars doesn't see into React components

### ISSUE THREAD

**Issue by hatched** [eslint/eslint#7418]
**Tell us about your environment**

* **ESLint Version:**
3.8.1
* **Node Version:**
6.2.0
* **npm Version:**
3.8.9

**What parser (default, Babel-ESLint, etc.) are you using?**
default

**Please show your full configuration:**

```
env:
  browser: true
  node: true
  es6: true
plugins:
  - react
parserOptions:
  ecmaVersion: 6
  ecmaFeatures:
    jsx: true
rules:
  no-unused-vars: [2, {args: "none"}]
```

**What did you do? Please include the actual source code causing the issue.**

```js
'use strict';

function test() {
  const components = window.foo.bar;
  return <components.baz/>;
};

test();
```

**What did you expect to happen?**
It should lint successfully.

**What actually happened? Please include the actual, raw output from ESLint.**
~/test.js
  4:9  error  'components' is assigned a value but never used      no-unused-vars

✖ 1 problems (1 errors, 0 warnings)

**Comment by platinumazure** [comment]
You need to enable rule `react/jsx-uses-vars`, I think.

**Comment by hatched** [comment]
That did it! Thanks @platinumazure
